cmd/mdatagen/internal/templates/feature_gates.go.tmpl

```diff
@@ -0,0 +1,26 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package {{ .Package }}
+
+import (
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+func init() {
+    for _, fg := range {{ .FeatureGates }} {
+    	var opts []featuregate.RegisterOption
+        if fg.FromVersion != "" {
+            opts = append(opts, featuregate.WithRegisterFromVersion(fg.FromVersion))
+        }
+        if fg.Description != "" {
+            opts = append(opts, featuregate.WithRegisterDescription(fg.Description))
+        }
+        if fg.ReferenceURL != "" {
+            opts = append(opts, featuregate.WithRegisterReferenceURL(fg.ReferenceURL))
+        }
+        if fg.ToVersion != "" {
+                opts = append(opts, featuregate.WithRegisterToVersion(fg.ToVersion ))
+        }
+        _ = featuregate.GlobalRegistry().MustRegister(fg.ID, fg.Stage, opts...)
+    }
+}
```

Comment:
I would prefer doing this with the templating instead of at runtime. You can have something like this

```go
opts = append(opts,
    {{if .from_version}} featuregate.WithRegisterFromVersion(.from_version),{{end}}
  // ...
)
```

see here for a more complicated templating example: https://github.com/open-telemetry/opentelemetry-collector/blob/403c782d50c630ad945feebe8b26e1f656ccd3c1/cmd/mdatagen/internal/templates/documentation.md.tmpl#L27-L31

Reply:
done